File: services/resource/src/main/java/com/github/exbotanical/resource/models/GouacheResponse.java
package com.github.exbotanical.resource.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class GouacheResponse {
  String internal;
  String friendly;
  Object data;
  int flags;
}

File: services/analytics/analytics/models/gouache_response.py
from flask import Response, jsonify


def gouache_response(data=None, friendly='', internal='', flags=0) -> Response:
    return jsonify(
        {'data': data, 'friendly': friendly, 'internal': internal, 'flags': flags}
    )


def ok_response(data: object) -> Response:
    return gouache_response(data)


def err_response(friendly: str, internal: str, flags=0) -> Response:
    return gouache_response(None, friendly, internal, flags)
